src/OpenSpaceToolkit/Astrodynamics/Estimator/OrbitDeterminationSolver.cpp

```diff
@@ -115,52 +114,50 @@ OrbitDeterminationSolver::Analysis OrbitDeterminationSolver::estimate(
                                                 : anEstimationCoordinateSubsets
     );
 
-    // Validate estimator subsets
+    // Validate subsets
     for (const auto& subset : estimationStateBuilder.getCoordinateSubsets())
     {
         if (!initialGuessStateInEstimationFrame.hasSubset(subset))
         {
             throw ostk::core::error::RuntimeError(
-                "Input State must contain at least the coordinate subsets that we want to estimate."
+                "InitialGuessState must contain at least all of the EstimationCoordinateSubsets."
             );
         }
     }
 
-    // Define state generation callback
-    auto generateStatesCallback = [&](const State& aState, const Array<Instant>& anInstantArray) -> Array<State>
+    const auto aStateGenerator = [&](const State& aState, const Array<Instant>& anInstantArray) -> Array<State>
     {
         const State propagatorState = propagationStateBuilder.expand(aState, initialGuessStateInEstimationFrame);
         return propagator_.calculateStatesAt(propagatorState, anInstantArray);
     };
 
-    // Solve least squares problem
-    const LeastSquaresSolver::Analysis analysis = solver_.solve(
+    LeastSquaresSolver::Analysis analysis = solver_.solve(
         estimationStateBuilder.reduce(initialGuessStateInEstimationFrame),
-        observationsInEstimationFrame,
-        generateStatesCallback,
+        observationStatesInEstimationFrame,
+        aStateGenerator,
         anInitialGuessSigmas,
         anObservationSigmas
     );
 
-    // Expand solution state to full state
     const State estimatedState =
         propagationStateBuilder.expand(analysis.estimatedState, initialGuessStateInEstimationFrame)
-            .inFrame(anInitialGuessState.accessFrame());
+            .inFrame(estimationFrameSPtr_);
+    analysis.estimatedState = estimatedState;
```

Comment:
We don't want to update this in my opinion. This state is what was solved by the least squares solver, in it's frame, and coordinates which could have been reduced etc.

Reply:
I don't think it's that complicated, the estimated state is at the OD analysis level, and the remaining one is at the lease squares solver level. You don't want to lose the information of what the solver did today it's a LeastSquaresSolver tomorrow it might be a UKF